### PR TITLE
add `allowDeclarationsAfterNestedRules` option

### PR DESCRIPTION
**Pull request by romainmenke**
- updated README with new option
- added type comment for postcss 8
- added tests for the new option specifically

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "postcss-nesting",
   "description": "Nest rules inside each other in CSS",
   "license": "CC0-1.0",
-  "version": "9.0.1",
+  "version": "10.0.0",
```
Comment (Antonio-Laguna) [inline package.json:5]: I'd revert this change since `npm version major` will take care of this and the tagging along

**Review by Antonio-Laguna — COMMENTED** [review]
Just one minor thing @romainmenke !

Great work again.

I don't see that we've dropped the support for PostCSS 7 as discussed. Is that meant to come later? Would you like me taking care of that? I think this will also simplify the testing moving forward.

File: package.json (hunk 1 above)
Comment (romainmenke) [inline package.json:5]:
```suggestion
  "version": "9.0.1",
```

**Comment by romainmenke** [comment]
> I don't see that we've dropped the support for PostCSS 7 as discussed. Is that meant to come later? Would you like me taking care of that? I think this will also simplify the testing moving forward.

This should be separate (imho) because one doesn't really have much to do with the other :)

**Review by Antonio-Laguna — APPROVED** [review]
This LGTM 👍

**Review by jonathantneal — APPROVED** [review]
